### PR TITLE
Custom plugins doc

### PR DESCRIPTION
**Pull request by lenhattan86**
Added `docs/features/custom-plugins.md`
Updated `docs/README.md`

### CODE REVIEW

**Review by ridv — APPROVED** [review]

**Review by mauri — COMMENTED** [review]
nice! added a few comments to state the purpose of the plugin more explicitly.

**Comment by ridv** [comment]
> nice! added a few comments to state the purpose of the plugin more explicitly.

Awesome! I'll make sure they get added 😃

File: docs/features/custom-plugins.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+Custom Plugins
+==============
+
+OfferSet
+--------
+OfferSet holds all offers for the scheduler. OfferSet is customizable via Interface `org.apache.aurora.scheduler.offers.OfferSet`.
+This interface is injectable via the '-offer_set_module' flag.
```
Comment (mauri) [inline docs/features/custom-plugins.md:7]: The purpose of "OfferSet" as a plugin is a bit elusive here. Could you add a few sentences about its purpose and how it relates with other offer related parameters in the scheduler? (i.e. be able to have more flexibility in the filtering/ordering of the offers when offer_order is insufficient?).

Additionally given that the OfferSet interface is a bit lacking on documentation, add a few notes here about its life cycle. I.e. when add(), remove(), or getOrdered() are invoked by the scheduler; and how is that related with offer_filter_duration, min_offer_hold_time, etc.

Comment (lenhattan86) [reply docs/features/custom-plugins.md:7]: will do some research and add them. thanks @mauri

Comment (ridv) [reply docs/features/custom-plugins.md:7]: Maybe we can borrow some words from the original proposal: https://docs.google.com/document/d/1fVHLt9AF-YbOCVCDMQmi5DATVusn-tqY8DldKbjVEm0/edit

**Pull request merged**